### PR TITLE
feat: convert spelling check to a standard check targets

### LINKED ISSUE #115: 🛠️ [TASK] : Convert Spelling check to a standard `check` targets.
### Summary

Make spelling check conform to check target pattern

### Description

Currently Spelling checks are independent of our standard CI pipeline.

We still need the ability to efficiently run the check locally.

But in CI we can use a standard check target for these and then remove the github action which is triggering these.

This will help better align our CI pipeline and reduce the number of github actions we need to enforce.

We would also need to add Spelling guide, which is be a part of this ticket.
This ticket also includes updating both:
https://github.com/input-output-hk/catalyst-voices
https://github.com/input-output-hk/hermes

To use the update check target, each will need a new issue raised to track this issue, and for the PR in those repos to be raised against that Issue.

### PR DESCRIPTION
**Pull request by apskhem**
# Description

## Related Issue(s)

> Closes #115 

## Description of Changes

Make spelling check conform to check target pattern

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: docs/src/guides/spelling.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+It can be used to check for misspelled words, identify potential errors, and suggest corrections.
+
+## Configuration
+
+Each repo will need a [`cspell.json`](http://cspell.org/configuration/) file in the root of the repo.
+This file configures `cspell`.
+The file provided in the `Catalyst-CI` repo should be taken as a starting point
+for new projects.
+
+## Project Words
```
Comment (Mr-Leshiy) [inline docs/src/guides/spelling.md:22]: I would also add a note that it is possible to extend this list of words for each specific file, which we are broadly using.
```
cspell: words <words>
```

Examples :
- https://github.com/input-output-hk/catalyst-ci/blob/c2f7d03c6c7c7ed5227e3398efab4b1d6c7d1508/earthly/docs/Earthfile#L3
- https://github.com/input-output-hk/catalyst-ci/blob/c2f7d03c6c7c7ed5227e3398efab4b1d6c7d1508/docs/src/guides/languages/postgresql.md#L12

File: Earthfile [hunk 2]
```diff
@@ -16,12 +16,20 @@ markdown-check-fix:
 
     DO ./earthly/mdlint+MDLINT_LOCALLY --src=$(echo ${PWD}) --fix=--fix
 
+# spell-check Check spelling in this repo locally.
 spell-check:
```
Comment (Mr-Leshiy) [inline Earthfile:20]: let's rename this target to `local-spell-check`

File: Earthfile [hunk 3]
```diff
@@ -16,12 +16,20 @@ markdown-check-fix:
...
-    # Check spelling in this repo.
     LOCALLY
 
     DO ./earthly/cspell+CSPELL_LOCALLY --src=$(echo ${PWD})
 
+# check-spelling Check spelling in this repo inside a container.
+check-spelling:
```
Comment (Mr-Leshiy) [inline Earthfile:26]: rename it to just `spell-check`

File: docs/src/guides/spelling.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
```
Comment (Mr-Leshiy) [inline docs/src/guides/spelling.md:1]: Also provide an Earthly example how to play with spell checking.
You can just copy that code which you have already written with `check-spelling` and `check` target

File: earthly/cspell/Earthfile [hunk 5]
```diff
@@ -24,10 +24,28 @@ CSPELL_LOCALLY:
         ghcr.io/streetsidesoftware/cspell:8.0.0 \
         lint . --dot
 
+CHECK:
```
Comment (Mr-Leshiy) [inline earthly/cspell/Earthfile:27]: Let's rename `CSPELL_LOCALLY` to `LOCAL_CHECK` or `CHECK_LOCALLY`

**Review by Mr-Leshiy — CHANGES_REQUESTED** [review]

**Comment by Mr-Leshiy** [comment]
BTW, I think we should add this `check` target to the separate Earthfile, for example to the `earthly/cspell/Earthfile`. 
So the idea is to run spell checks and markdown checks separately in different job, rather than in one as we will put it under the similar `check` target inside the root `Earthfile`. 
Because if we will put it under the same target like this
```
check:
    BUILD +check-spelling
    BUILD +check-markdown
```
if spell check will fail, markdown check will not even run, what we don't want

@stevenj

File: docs/src/guides/spellcheck.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+## Introduction
+
+checking is integrated into the `check` pipeline.
+The reference to the pipeline can be found [here](https://input-output-hk.github.io/catalyst-ci/onboarding/).
+The goal of the `check` stage is to ensure the overall health of the project.
+
+This utilizes [`cspell`](https://cspell.org) under the hood for checking code and other text documents.
+It can be used to check for misspelled words, identify potential errors, and suggest corrections.
+
+## Using the spell checking
```
Comment (Mr-Leshiy) [inline docs/src/guides/spellcheck.md:16]: Also don't forget to add about local spell checking run

**Review by stevenj — CHANGES_REQUESTED** [review]

**Review by stevenj — APPROVED** [review]
LGTM

**Pull request merged**